database/SCHEMA.md

```diff
@@ -0,0 +1,229 @@
+| ---------- | ------------ |
+| TxID (u64) | Height (u64) |
+
+`Constant Size = true`
+
+### Tx Unlock Time
+
+This table will store the unlock time of a tx (only if the tx has a non-zero lock-time).
+
+| Key        | Value             |
+| ---------- | ----------------- |
+| TxID (u64) | unlock time (u64) |
+
+`Constant Size = true`
+
+### Pruned Tx Blobs
+
+This table will contain pruned tx blobs (even if the DB is not pruned).
+
+| Key        | Value                 |
+| ---------- | --------------------- |
+| TxID (u64) | Pruned Blob (Vec<u8>) |
+
+`Constant Size = false`
+
+### Prunable Tx Blobs
+
+This table will contain the prunable part of a tx.
+
+| Key        | Value                   |
+| ---------- | ----------------------- |
+| TxID (u64) | Prunable Blob (Vec<u8>) |
+
+`Constant Size = false`
+
+### Prunable Hash
+
+This table will contain the hash of the prunable part of a tx.
+
+| Key        | Value                    |
+| ---------- | ------------------------ |
+| TxID (u64) | Prunable Hash ([u8; 32]) |
+
+`Constant Size = true`
+
+## Tx Outputs
+
+This table gives the amount idxs of the transaction outputs.
+
+| Key        | Value                  |
+| ---------- | ---------------------- |
+| TxID (u64) | Amount Idxs (Vec<u64>) |
+
+`Constant Size = false`
+
+## Outputs
+
+### CryptoNote Outputs
+
+This table will contain legacy CryptoNote outputs which have clear amounts.
+This table will not contain an output with 0 amount.
+
+| Primary Key  | Secondary Key      | Value              |
+| ------------ | ------------------ | ------------------ |
+| Amount (u64) | Amount Index (u32) | Output (See below) |
+
+> This table stores the amount idex as a u32 to save space as the creation of v1 txs is limited, u32::MAX should never be hit.
+
+```rust
+struct Output{
+    key: [u8; 32],
+    // We could get this from the tx_idx with the Tx Heights table but that would require another look up per out.
+    height: u64, 
+    tx_idx: u64,
+    // For if the tx that created this out has a time-lock - this means we only need to look on Tx Unlock Time if this is true.
+    locked: bool 
+}
+// TODO: local_index?
+
+```
+
+`Constant Size = true`
+
+### RingCT Outputs
+
+| Key                | Value                 |
+| ------------------ | --------------------- |
+| Amount Index (u64) | RctOutput (see below) |
+
+```rust
+struct RctOutput{
+    key: [u8; 32],
+    // We could get this from the tx_idx with the Tx Heights table but that would require another look up per out.
+    height: u64, 
+    tx_idx: u64,
+    // For if the tx that created this out has a time-lock - this means we only need to look on Tx Unlock Time if this is true.
+    locked: bool,
+    // The amount commitment of this output.
+    commitment: [u8; 32]
+}
+// TODO: local_index?
+
+```
+
+`Constant Size = true`
+
+> TODO: We could split this table again into `RingCT (non-miner) Outputs` and `RingCT (miner) Outputs` as for miner outputs we can
```

Comment:
it means we would need to lookup an output twice to make sure we don't have it though and it would mean miner outputs require 2 look ups, provided we look in `non-miner` first.

This is an optimization we should probably do later to asses the performance impact.